File: app/src/main/java/calendario/kevshupp/diariokevinali/LastMessageLargeWidget.java
package calendario.kevshupp.diariokevinali;

import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;

public class LastMessageLargeWidget extends AppWidgetProvider {

    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        for (int appWidgetId : appWidgetIds) {
            LastMessageWidget.updateAppWidget(context, appWidgetManager, appWidgetId, true, false);
        }
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        super.onReceive(context, intent);
    }
}

File: app/src/main/java/calendario/kevshupp/diariokevinali/LastMessageLargeWidget.kt
package calendario.kevshupp.diariokevinali

import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.Context
import android.content.Intent

class LastMessageLargeWidget : AppWidgetProvider() {

    override fun onUpdate(context: Context, appWidgetManager: AppWidgetManager, appWidgetIds: IntArray) {
        for (appWidgetId in appWidgetIds) {
            LastMessageWidget.updateAppWidget(context, appWidgetManager, appWidgetId, forceLarge = true, forceSmall = false)
        }
    }

    override fun onReceive(context: Context, intent: Intent) {
        super.onReceive(context, intent)
    }
}
